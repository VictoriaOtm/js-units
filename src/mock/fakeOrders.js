export const fakeOrders = [
	{
		id: 123,
		date: 1544356800000,
		shop: 'Ali Express',
		items: [
			'Утиный пластмасса для показ новый год',
			'Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан',
			'Новый стиль один розница яйцо для упаковки форма латекс',
		]
	},
	{
		id: 124,
		date: 1552481120000,
		shop: 'Lamoda.ru',
		items: [
			'Жакет - BOREAL5',
			'Miss Gabby Костюм',
			'Ostin перчатки мужские',
			'Zara худи роз.',
		]
	},
	{
		id: 126,
		date: 1552585550000,
		shop: 'Эльдорадо',
		items: [
			'Ноутбук Apple MacBook Air 13.3" (MQD32RU/A)',
			'Игровая приставка Sony PlayStation 4 Pro 1TB Black (CUH-7208B)',
		]
	},
	{
		id: 127,
		date: 1552585550000,
		shop: 'Эльдорадо2',
		items: []
	}
];
