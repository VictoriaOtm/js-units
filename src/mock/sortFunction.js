export const sortFunctionMock = jest.fn(() => {
    return 1;
})
