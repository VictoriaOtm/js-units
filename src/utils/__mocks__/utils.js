export const fakeDate = 1552481120000;
