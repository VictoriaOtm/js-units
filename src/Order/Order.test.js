import {fakeOrders} from "../mock/fakeOrders";

jest.mock('../utils/getDate');
import {getDate} from '../utils/getDate';

import React from 'react';
import Order from './Order';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Order component', () => {
    const testDate = '123date123'
    let wrapper = shallow(<Order/>);

    beforeAll(() => {
        getDate.mockReturnValue(testDate);
    })

    beforeEach(() => {
        wrapper = shallow(<Order/>);
    })

    afterAll(() => {
        jest.resetAllMocks();
    })

    it('default', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('non-empty order', () => {
        wrapper = shallow(<Order order = {fakeOrders[0]}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('order with empty items', () => {
        wrapper.setProps({order: {shop: "shop", date: getDate(), items: []}});

        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
