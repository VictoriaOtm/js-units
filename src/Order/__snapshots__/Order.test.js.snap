// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render with default state DATE 1`] = `""`;

exports[`Order component default 1`] = `""`;

exports[`Order component non-empty order 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span
      id="date"
    >
      123date123
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component order with empty items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      123date123
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with default state DATE 1`] = `""`;

exports[`Order component test 1`] = `""`;
